packages/react-components/react-tabster/src/hooks/useTabster.ts

```diff
@@ -34,15 +32,51 @@ export const useTabster = (): TabsterTypes.TabsterCore | null => {
         !!element.firstElementChild?.hasAttribute('data-is-focus-trap-zone-bumper'),
       DOMAPI: shadowDOMAPI,
     });
-  }, [defaultView, shadowDOMAPI]);
+  }
+}
+
+/**
+ * Tries to get a tabster instance on the current window or creates a new one
+ * Since Tabster is single instance only, feel free to call this hook to ensure Tabster exists if necessary
+ *
+ * @internal
+ * @returns Tabster a ref to core instance or a factory result
+ */
+export function useTabster(): React.RefObject<TabsterTypes.TabsterCore | null>;
+export function useTabster<FactoryResult>(
+  factory: UseTabsterFactory<FactoryResult>,
+): React.RefObject<FactoryResult | null>;
+
+export function useTabster<FactoryResult>(factory = DEFAULT_FACTORY) {
+  const { targetDocument } = useFluent();
+  const factoryRef = React.useRef<FactoryResult | null>(null);
```

Comment:
nit: shouldn't this be named `factoryResultRef`?! 👀, I mean... it's not a ref to the factory....

Reply:
true, will update 👍